docs/Examples/Upload.example.purs

```diff
@@ -35,18 +37,21 @@ component = createComponent "UploadExample"
 data Action
   = NoOp
   | SetReadonly Boolean
-  | ImageEx (Array FileId)
-  | FileEx (Array FileId)
-  | AvatarEx (Maybe FileId)
+  | NewImages (Array FileId)
+  | RemoveImage FileId
+  | NewFiles (Array FileId)
+  | RemoveFile FileId
+  | NewAvatar FileId
+  | RemoveAvatar
   | StartUpload
   | InitializeUploadBuffer (AVar Unit)
 
 docs :: JSX
 docs = unit # make component { initialState, didMount, render }
   where
     initialState =
-      { images: []
-      , files: []
+      { images: Set.empty
```

Comment:
I'd actually be tempted to leave these as arrays, so that files continue to be displayed the order they were selected/uploaded. If we use a Set here, will they be displayed in a random order (or at least an order which appears to be random) in the UI?

Reply:
Oh yeah, good point. I'll fix that, thanks! Just changed it to `Set` so that the potential deduping would be handled automatically.